lib/compat/wordpress-6.2/class-gutenberg-rest-pattern-directory-controller-6-2.php

```diff
@@ -10,6 +10,125 @@
  * Controller which provides REST endpoint for block patterns from wordpress.org/patterns.
  */
 class Gutenberg_REST_Pattern_Directory_Controller_6_2 extends Gutenberg_REST_Pattern_Directory_Controller_6_0 {
+	/**
+	 * Registers the necessary REST API routes.
+	 *
+	 * @since 5.8.0
+	 * @since 6.2.0 Added pattern directory categories endpoint.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/categories',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_pattern_categories' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				),
+			)
+		);
+
+		parent::register_routes();
+	}
+
+	/**
+	 * Retrieve block patterns categories.
+	 *
+	 * @since 6.2.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_pattern_categories( $request ) {
+		$api_url       = 'http://api.wordpress.org/patterns/1.0/?categories';
+		$transient_key = 'wp_remote_block_pattern_categories_' . md5( $api_url );
+
+		/**
+		 * Use network-wide transient to improve performance. The locale is the only site
+		 * configuration that affects the response, and it's included in the transient key.
```

Comment:
This isn't true for this request. Do you know if the categories endpoint supports the locale?

Reply:
You're right about the wrong comment and that's why I'm adding to the transient key only the url above. I don't think the categories support `locale` though.. -cc @ryelle

Reply:
I think it does. See `http://api.wordpress.org/patterns/1.0/?categories&locale=es_ES`.

Reply:
It's the same names for me with the above request as well.. 🤔

Reply:
This is what I see when accessing the URL via the browser.

![CleanShot 2022-11-15 at 17 00 08](https://user-images.githubusercontent.com/240569/201925867-2e0a9578-2584-4544-bacd-d8330d637d12.png)